pkg/math/math.go

```diff
@@ -51,3 +52,33 @@ func BigFloatToBigInt(f *big.Float, decimals uint64) *big.Int {
 
 	return result
 }
+
+// Float64StringToBigFloat converts a float64 string to a big.Float.
+func Float64StringToBigFloat(s string, decimals uint64) (*big.Float, error) {
+	bigFloat := new(big.Float)
+	_, ok := bigFloat.SetString(s)
+	if !ok {
+		return nil, fmt.Errorf("failed to set big.Float from string: %s", s)
+	}
+
+	return ScaleBigFloat(bigFloat, decimals), nil
+}
+
+// Float64ToBigFloat converts a float64 to a big.Float.
+func Float64ToBigFloat(val float64, decimals uint64) *big.Float {
+	bigVal := new(big.Float)
+	bigVal.SetFloat64(val)
+
+	return ScaleBigFloat(bigVal, decimals)
+}
+
+// ScaleBigFloat scales a big.Float by the given decimals.
+func ScaleBigFloat(f *big.Float, decimals uint64) *big.Float {
```

Comment:
test

Reply:
Why do we need this? Like the entire purpose of using `big.Floats` is that they don't need to be scaled for arithmetic?